Resolve login request with an object containing hasGrantedScopes status.

deCodeIt opened the pull request:
Resolves https://github.com/vue-gapi/vue-gapi/issues/41.

Currently, there's no way in this API to let the user know whether the requested scopes were granted or not. This PR consists of changes required to return the hasGrantedScopes  status with onResolve callback.

namoscato reviewed (commented):
@deCodeIt, this seems like a solid approach to me and resolves https://github.com/vue-gapi/vue-gapi/issues/41! Let me know when it's good for a review, and I can test this out.

deCodeIt commented:
> @deCodeIt, this seems like a solid approach to me and resolves #41! Let me know when it's good for a review, and I can test this out.

@namoscato It's ready for a review.

deCodeIt commented:
Tagging @namoscato for review

namoscato reviewed (commented):
This is looking great, @deCodeIt!

I'll run through some manual testing by this weekend; thanks for your patience. _(Sidenote: We should throw an example in this repository or do something to make it easier to actually test changes.)_

deCodeIt commented on .babelrc:
```diff
@@ -14,6 +14,7 @@
   "env": {
     "test": {
       "plugins": [
+        "@babel/transform-runtime",
```
This plugin is added because the newly added **async** `jest` test was failing due to its absence.
As tests are for developers, hence added this dependency as a devDependency.

deCodeIt commented on test/GoogleAuthService.test.js:
```diff
@@ -13,3 +14,52 @@ it('_expiresAt returns string of numbers only', () => {
+  gAuth.currentUser.resolveGrant()
+  res = await newService.grant()
+  expect(res).toBeInstanceOf(GoogleUser)
+})
```
Yes, I felt the need for some level of testing hence added this. Although this may or may not be the best possible way to write a test at the same time this is what came to my mind after a brainstorming session.

1. Yes adding mock to the name sounds good. Will do
2. Sure, makes sense.

deCodeIt commented:
> This is looking great, @deCodeIt!
> 
> I'll run through some manual testing by this weekend; thanks for your patience. _(Sidenote: We should throw an example in this repository or do something to make it easier to actually test changes.)_

@namoscato Addressed your comments

deCodeIt commented:
> > This is looking great, @deCodeIt!
> > I'll run through some manual testing by this weekend; thanks for your patience. _(Sidenote: We should throw an example in this repository or do something to make it easier to actually test changes.)_
> 
> @namoscato Addressed your comments

I've now added test for login as well.

namoscato reviewed (changes requested):
Just one small suggestion, but I tested this out, and it's working great!

deCodeIt commented on src/VueGapi/GoogleAuthService.js:
```diff
@@ -140,6 +141,29 @@ export default class GoogleAuthService {
+  /**
+   * @typedef LoginResponse
+   * @property {bool} hasGrantedScopes True if the requested scopes were granted.
+   * @property {GoogleUser} gUser GoogleUser
```
Yes, googleUser sounds great, will do

deCodeIt commented:
> Just one small suggestion, but I tested this out, and it's working great!

Addressed the above comment.
Looks like we can merge this now.
